data/net/minecraft/world/entity/ai/util/RandomPos.mapping

```diff
@@ -0,0 +1,47 @@
+CLASS net/minecraft/world/entity/ai/util/RandomPos
+	METHOD generateRandomDirection (Ljava/util/Random;II)Lnet/minecraft/core/BlockPos;
+		COMMENT Gets a random position within a certain distance.
+		ARG 0 random
+		ARG 1 horizontalDistance
+		ARG 2 verticalDistance
+	METHOD generateRandomDirectionWithinRadians (Ljava/util/Random;IIIDDD)Lnet/minecraft/core/BlockPos;
+		COMMENT @return a random (x, y, z) coordinate by picking a point (x, z), adding a random angle, up to a difference of maxAngleDelta. The y position is randomly chosen from the range [y - yRange, y + yRange]. Will be null if the chosen coordinate is outside of a distance of maxHorizontalDistance from the origin.
+		ARG 0 random
+		ARG 1 maxHorizontalDifference
+			COMMENT The maximum value in x and z, in absolute value, that could be returned.
+		ARG 2 yRange
+			COMMENT The range plus or minus the y position to be chosen
+		ARG 3 y
+			COMMENT The target y position
+		ARG 4 x
+			COMMENT The x offset to the target position
+		ARG 6 z
+			COMMENT The z offset to the target position
+		ARG 8 maxAngleDelta
+			COMMENT The maximum variance of the returned angle, from the base angle being a vector from (0, 0) to (x, z).
+	METHOD generateRandomPos (Ljava/util/function/Supplier;Ljava/util/function/ToDoubleFunction;)Lnet/minecraft/world/phys/Vec3;
+		COMMENT Tries 10 times to maximize the return value of the position to double function based on the supplied position
+		ARG 0 posSupplier
+		ARG 1 toDoubleFunction
+	METHOD generateRandomPos (Lnet/minecraft/world/entity/PathfinderMob;Ljava/util/function/Supplier;)Lnet/minecraft/world/phys/Vec3;
+		ARG 0 mob
+		ARG 1 posSupplier
+	METHOD generateRandomPosTowardDirection (Lnet/minecraft/world/entity/PathfinderMob;ILjava/util/Random;Lnet/minecraft/core/BlockPos;)Lnet/minecraft/core/BlockPos;
+		COMMENT @return a random position within range, only if the mob is currently restricted
+		ARG 0 mob
+		ARG 1 range
+		ARG 2 random
+		ARG 3 pos
+	METHOD moveUpOutOfSolid (Lnet/minecraft/core/BlockPos;ILjava/util/function/Predicate;)Lnet/minecraft/core/BlockPos;
+		COMMENT @return the highest above position that is within the provided conditions
+		ARG 0 pos
+		ARG 1 maxY
+		ARG 2 posPredicate
+	METHOD moveUpToAboveSolid (Lnet/minecraft/core/BlockPos;IILjava/util/function/Predicate;)Lnet/minecraft/core/BlockPos;
+		COMMENT Finds a position above based on the conditions.
+		COMMENT
+		COMMENT After it finds the position once, it will continue to move up until aboveSolidAmount is reached or the position is no longer valid
```

Comment:
```suggestion
		COMMENT <p>After it finds the position once, it will continue to move up until {@code aboveSolidAmount} is reached or the position is no longer valid.
```